src/utils/generator.rs

```diff
@@ -0,0 +1,216 @@
+//! A number of methods to generate random data for testing.
+
+use crate::types::{location::Location, node::Node, status};
+use ordered_float::OrderedFloat;
+use quaternion::Quaternion;
+use rand::{rngs::ThreadRng, Rng};
+use uuid::Uuid;
+use vecmath::Vector3;
+
+//-----------------------------------------------------
+// Constants
+//-----------------------------------------------------
+const DEG_TO_RAD: f32 = std::f32::consts::PI / 180.0;
+const RAD_TO_DEG: f32 = 180.0 / std::f32::consts::PI;
+
+/// Generate a vector of random nodes.
+///
+/// # Arguments
+/// * `capacity` - The number of nodes to generate.
+///
+/// # Returns
+/// A vector of nodes.
+pub fn generate_nodes(capacity: i32) -> Vec<Node> {
+    let mut nodes = Vec::new();
+    for _ in 0..capacity {
+        nodes.push(generate_random_node());
+    }
+    nodes
+}
+
+/// Generate a vector of random nodes near a location.
+///
+/// # Arguments
+/// * `location` - The location to generate nodes near.
+/// * `radius` - The radius in kilometers to generate nodes within.
+/// * `capacity` - The number of nodes to generate.
+///
+/// # Returns
+/// A vector of nodes.
+pub fn generate_nodes_near(location: &Location, radius: f32, capacity: i32) -> Vec<Node> {
+    let mut nodes = Vec::new();
+    for _ in 0..capacity {
+        nodes.push(generate_random_node_near(location, radius));
+    }
+    nodes
+}
+
+/// Generate a single random node.
+pub fn generate_random_node() -> Node {
+    Node {
+        uid: Uuid::new_v4().to_string(),
+        location: generate_location(),
+        forward_to: None,
+        status: status::Status::Ok,
+    }
+}
+
+/// Generate a random node near a location within radius in kilometers.
+///
+/// # Arguments
+/// * `location` - The location to generate nodes near.
+/// * `radius` - The radius in kilometers to generate nodes within.
+///
+/// # Returns
+/// A node with a location near the given location.
+pub fn generate_random_node_near(location: &Location, radius: f32) -> Node {
+    Node {
+        uid: Uuid::new_v4().to_string(),
```

Comment:
Regarding UUID:
> The uniqueness property is not strictly guaranteed, however for all practical purposes, it can be assumed that an unintentional collision would be extremely unlikely.

Should still have a check that there are no clashing UUIDs

Reply:
The checking should be left for the users to perform because this function only generate one node, and there's no global state for the function keep track of an existing set of generated UUIDs. 

But I can include some caution notes in the doc comment.

Reply:
See fix in [acb59d6](https://github.com/Arrow-air/lib-router/pull/6/commits/acb59d614bb507057a9b9cdd797b943a650b415d)

Reply:
Thank you